src/modules/vtol_att_control/standard.cpp

```diff
@@ -337,78 +337,78 @@ void Standard::update_fw_state()
+		fw_out[actuator_controls_s::INDEX_YAW]          = 0;
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = _pusher_throttle;
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = 0;
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = 0;
 
-	// set the fixed wing throttle control
-	if (_vtol_schedule.flight_mode == vtol_mode::FW_MODE) {
+		break;
 
-		// take the throttle value commanded by the fw controller
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] =
-			_actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];
+	case vtol_mode::TRANSITION_TO_FW:
 
-	} else {
-		// otherwise we may be ramping up the throttle during the transition to fw mode
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] = _pusher_throttle;
-	}
+	// FALLTHROUGH
+	case vtol_mode::TRANSITION_TO_MC:
+		// MC out = MC in (weighted)
+		mc_out[actuator_controls_s::INDEX_ROLL]         = mc_in[actuator_controls_s::INDEX_ROLL]     * _mc_roll_weight;
+		mc_out[actuator_controls_s::INDEX_PITCH]        = mc_in[actuator_controls_s::INDEX_PITCH]    * _mc_pitch_weight;
+		mc_out[actuator_controls_s::INDEX_YAW]          = mc_in[actuator_controls_s::INDEX_YAW]      * _mc_yaw_weight;
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = mc_in[actuator_controls_s::INDEX_THROTTLE] * _mc_throttle_weight;
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = 0;
+
+		// FW out = FW in, with VTOL transition controlling throttle and airbrakes
+		fw_out[actuator_controls_s::INDEX_ROLL]         = fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = fw_in[actuator_controls_s::INDEX_YAW];
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = _pusher_throttle;
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = 0;
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = _reverse_output;
+
+		break;
 
+	case vtol_mode::FW_MODE:
+		// MC out = 0
+		mc_out[actuator_controls_s::INDEX_ROLL]         = 0;
+		mc_out[actuator_controls_s::INDEX_PITCH]        = 0;
+		mc_out[actuator_controls_s::INDEX_YAW]          = 0;
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = 0;
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = 0;
+
+		// FW out = FW in
+		fw_out[actuator_controls_s::INDEX_ROLL]         = fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = fw_in[actuator_controls_s::INDEX_YAW];
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = fw_in[actuator_controls_s::INDEX_THROTTLE];
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = fw_in[actuator_controls_s::INDEX_FLAPS];
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = fw_in[actuator_controls_s::INDEX_AIRBRAKES];
+		break;
+	}
 
+	_actuators_out_0->timestamp = _actuators_out_1->timestamp = hrt_absolute_time();
```

Comment:
`timestamp_sample` seems to be missing for the standard VTOL.

Reply:
The initial memcpy captures it, but now I'm thinking it might be better off to copy each field explicitly (including timestamp_sample, landing gear, etc).

Reply:
Updated.